src/ec/curve25519/ops.rs

```diff
@@ -121,6 +121,72 @@ fn encode_point(x: Elem, y: Elem, z: Elem) -> EncodedPoint {
     bytes
 }
 
+#[cfg(all(target_arch = "x86_64", not(windows)))]
+pub fn scalar_mult_with_base_point(scalar: &Scalar) -> EncodedPoint {
+    let mut point = [0u8; ELEM_LEN];
+
+    unsafe { GFp_x25519_scalar_mult(&mut point, scalar, &BASE_POINT); }
+
+    point
+}
+
+#[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+pub fn scalar_mult_with_base_point(scalar: &Scalar) -> EncodedPoint {
+    if is_neon_capable() {
+        let mut result: EncodedPoint = [0; ELEM_LEN];
+        unsafe { GFp_x25519_NEON(&mut result, scalar, point); }
+        return result;
+    }
+
+    scalar_mult_with_base_point_generic(scalar)
+}
+
+#[cfg(any(all(target_arch = "x86_64", not(windows)),
+          any(target_arch = "arm", target_arch = "aarch64")))]
+const BASE_POINT: EncodedPoint = [
+    9, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0,
+];
+
+#[cfg(not(any(all(target_arch = "x86_64", not(windows)),
+              any(target_arch = "arm", target_arch = "aarch64"))))]
+pub fn scalar_mult_with_base_point(scalar: &Scalar) -> EncodedPoint {
+    scalar_mult_with_base_point_generic(scalar)
+}
+
+#[cfg(not(all(target_arch = "x86_64", not(windows))))]
+fn scalar_mult_with_base_point_generic(scalar: &Scalar) -> EncodedPoint {
+    let mut point = ExtPoint::new_at_infinity();
+
+    let mut masked = [0u8; SCALAR_LEN];
+    masked.copy_from_slice(scalar);
+    unsafe {
+        GFp_x25519_scalar_mask(&mut masked);
+        GFp_x25519_ge_scalarmult_base(&mut point, masked);
+    }
+
+    let mut z_plus_y = [0i32; ELEM_LIMBS];
+    let mut z_minus_y = [0i32; ELEM_LIMBS];
+    let mut z_minus_y_inv = [0i32; ELEM_LIMBS];
+    let mut coord_elem = [0i32; ELEM_LIMBS];
+    unsafe {
+        GFp_fe_add(&mut z_plus_y, &a.z, &a.y);
+        GFp_fe_sub(&mut z_minus_y, &a.z, &a.y);
+        GFp_fe_invert(&mut z_minus_y_inv, &z_minus_y);
+        GFp_fe_mul(&mut coord_elem, &z_plus_y, &z_minus_y_inv);
+    }
+    let encoded_point = [0u8; ELEM_LEN];
+    encoded_point.copy_from_slice(&coord_elem);
```

Comment:
Yeah, so actually looking into them and comparing against the RFCs, there only needs to be one *25519 scalar mask function, because they accomplish exactly the same thing (which is actually what I'd expect). I'm going to unify these.